cargo-shuttle/src/lib.rs

```diff
@@ -403,18 +406,78 @@ impl Shuttle {
     }
 
     async fn project_create(&self, client: &Client) -> Result<()> {
-        let project = client.create_project(self.ctx.project_name()).await?;
+        self.wait_with_spinner(
+            &[project::State::Ready],
+            Client::create_project,
+            self.ctx.project_name(),
+            client,
+        )
+        .await?;
 
-        println!("{project}");
+        Ok(())
+    }
+
+    async fn project_status(&self, client: &Client, follow: bool) -> Result<()> {
+        match follow {
+            true => {
+                self.wait_with_spinner(
+                    &[project::State::Ready, project::State::Destroyed],
+                    Client::get_project,
+                    self.ctx.project_name(),
+                    client,
+                )
+                .await?;
+            }
+            false => {
+                let project = client.get_project(self.ctx.project_name()).await?;
+                println!("{project}");
+            }
+        }
 
         Ok(())
     }
 
-    async fn project_status(&self, client: &Client) -> Result<()> {
-        let project = client.get_project(self.ctx.project_name()).await?;
+    async fn wait_with_spinner<'a, F, Fut>(
+        &self,
+        states_to_check: &[project::State],
+        f: F,
+        prj_name: &'a ProjectName,
```

Comment:
```suggestion
        project_name: &'a ProjectName,
```
Just to keep this aligned with the naming used else where